src/treq/response.py

```diff
@@ -95,3 +97,25 @@ def cookies(self):
                 jar.set_cookie(cookie)
 
         return jar
+
+    def fail_for_status(self):
```

Comment:
This strikes me as a somewhat awkward name.  Some alternatives:

  * `require_success`
  * `require_success_status`
  * `require_ok`
  * `check_status`
  * `check_success`

In my own use HTTP clients, I've often wanted to check for a _specific_ code.  For example, `CREATED` or `CONFLICT`.  However, maybe that's a check that can reasonable happen in a callback or errback on the Deferred returned by this method.